api/views.py

```diff
@@ -435,6 +436,46 @@ def handle_post(self, request, *args, **kwargs):
         return JsonResponse({'status': 'ok'})
 
 
+class ResendValidation(PublicJsonPostView):
+    def handle_post(self, request, *args, **kwargs):
+        body = json.loads(request.body.decode('utf-8'))
```

Comment:
We should avoid using `PublicJsonPostView` and instead try to use DRF for any API views.

Something like this https://github.com/IFRCGo/go-api/blob/c1e11c4f382384ac2b0ba7048474b149f2eecd31/api/views.py#L51-L62

Reply:
All the other similar views use the same pattern, like `ShowUsername` or `RecoverPassword`. Should I also change those to use `APIView` too?

Reply:
4847a24

Reply:
We would need to refactor so that all API views are using DRF views instead of the custom `PublicJsonPostView`.
It will be cool to follow this for any new addition to the `go-api`.
We planned to do this right? @batpad  @GregoryHorvath

Reply:
@thenav56 Since these are public Views we don't need these for the `PublicJsonPostView`s, right?
https://github.com/IFRCGo/go-api/blob/c1e11c4f382384ac2b0ba7048474b149f2eecd31/api/views.py#L52-L53

The part here is what confusing me, though it doesn't seem that we use this anywhere atm:
https://github.com/IFRCGo/go-api/blob/c1e11c4f382384ac2b0ba7048474b149f2eecd31/api/views.py#L283-L286

Reply:
@GregoryHorvath Yeah we don't need them. The best way will be to just define empty **permissions_classses**
```python3
permissions_classes = []
```
Basically we should use `request.user` directly instead of `get_authenticated_user`, and for any custom logic, we should create a custom authentication class and add it to DRF **DEFAULT_AUTHENTICATION_CLASSES**. But yeah it isn't used anywhere so we don't need to worry about it.